source/basic.tex

```diff
@@ -7269,3 +7339,443 @@
 the functions passed to \tcode{std::atexit()} or \tcode{std::at_quick_exit()}.%
 \indextext{program!termination|)}
 \indextext{program execution|)}
+
+\rSec1[basic.contract]{Contract assertions}%
+\indextext{contract assertion|(}%
+
+\rSec2[basic.contract.general]{General}%
+
+\pnum
+\defnx{Contract assertions}{contract assertion}
+allow the programmer to specify
+properties of the state of the program
+that are expected to hold at
+certain points during execution.
+Contract assertions are introduced by
+\grammarterm{precondition-specifier}s,
+\grammarterm{postcondition-specifier}s\iref{dcl.contract.func}, and
+\grammarterm{assertion-statement}s\iref{stmt.contract.assert}.
+
+\pnum
+Each contract assertion has a \defnadjx{contract-assertion}{predicate}{predicate},
+which is an expression of type \tcode{bool}.
+
+\begin{note}
+The value of the predicate is used to identify program states that are
+expected.
+\end{note}
+
+\pnum
+An invocation of the macro \tcode{va_start}\iref{cstdarg.syn}
+shall not be a subexpression
+of the predicate of a contract assertion,
+no diagnostic required.
+
+%JMB: This paragraph was not in P2900, but the note is not directly tied to the paragraph above.
+\pnum
+\begin{note}
+Within the predicate of a contract assertion,
+\grammarterm{id-expression}s referring to
+variables declared outside the contract assertion
+are const\iref{expr.prim.id.unqual}, %JMB: P2900 had this const in code font
+\tcode{this} is a pointer to const\iref{expr.prim.this}, %JMB: P2900 had this const in code font
```

Comment:
This is a note.  I agree non-code-font is slightly better (more casual, in a casual sentence).  Remove JMBs.

Reply:
done